tests/test_js_graph.py

```diff
@@ -1502,3 +1503,28 @@ def insert_by_parametrized_type(graph, cell_type, tags=None):
         assert False, "Unexpected cell type: {}".format(cell_type)
 
     return cell_id
+
+
+@pytest.mark.parametrize(
+    'layout_name',
+    [
+        QmxGraphApi.LAYOUT_ORGANIC,
+        QmxGraphApi.LAYOUT_COMPACT,
+        QmxGraphApi.LAYOUT_CIRCLE,
+        QmxGraphApi.LAYOUT_COMPACT_TREE,
+        QmxGraphApi.LAYOUT_EDGE_LABEL,
+        QmxGraphApi.LAYOUT_PARALLEL_EDGE,
+        QmxGraphApi.LAYOUT_PARTITION,
+        QmxGraphApi.LAYOUT_RADIAL_TREE,
+        QmxGraphApi.LAYOUT_STACK,
+    ],
+)
+def test_run_all_layouts(layout_name, graph_cases):
```

Comment:
I tried verifying the position before and after the run, but there are cases where they can be the same... I don't know a good way to test this :(

Leaving the (bad) test here for reference:

```python
@pytest.mark.parametrize(
    'layout_name',
    [
        QmxGraphApi.LAYOUT_ORGANIC,
        QmxGraphApi.LAYOUT_COMPACT,
        QmxGraphApi.LAYOUT_CIRCLE,
        QmxGraphApi.LAYOUT_COMPACT_TREE,
        QmxGraphApi.LAYOUT_EDGE_LABEL,
        QmxGraphApi.LAYOUT_PARALLEL_EDGE,
        QmxGraphApi.LAYOUT_PARTITION,
        QmxGraphApi.LAYOUT_RADIAL_TREE,
        QmxGraphApi.LAYOUT_STACK,
    ],
)
def test_run_all_layouts(layout_name, graph_cases):
    graph = graph_cases('3v_1e')
    label = lambda cell: graph.get_label(cell)
    nodes_positions = {
        label(v): {
            'before': None,
            'after': None,
        }
        for v in graph.get_vertices()
    }

    for v in graph.get_vertices():
        nodes_positions[label(v)]['before'] = graph.get_vertex_position(v)
    graph.eval_js_function('api.runLayout', layout_name)
    for v in graph.get_vertices():
        nodes_positions[label(v)]['after'] = graph.get_vertex_position(v)

    for position_data in nodes_positions.values():
        # We do not have the exact expected position to check - But we do know that the positions
        # should at least change.
        x_before, y_before = position_data['before']
        x_after, y_after = position_data['after']
        assert not pytest.approx(x_before) == x_after
        assert not pytest.approx(y_before) == y_after

Reply:
Not sure how to proceed here as well...

Reply:
A single test checking if organic layout works? (do you know others?)

How does the layout algorithms work with the decorations and/or tables?
Are they ignored? Moved?
What if the cell are relativelly positioned?

Reply:
**A single test checking if organic layout works? (do you know others?)**
Even the organic layout can return some cells in the same position, and it is stochastic (Results will change in each run). How can I check the results?

**How does the layout algorithms work with the decorations and/or tables? Are they ignored? Moved?**
We don't really care about decorations and tables... I didn't tested, but I don't care if they are moved around.

**What if the cell are relativelly positioned?**
I didn't even know this is possible. How can this be checked?

Reply:
Ah... also note that `3v_1e` has just 1 edge having an disconnected vertex and disconected vertices are not moved.

Reply:
> (Results will change in each run). How can I check the results?
In you test you could mock the `Math.random` to return a fixed value yielding deterministic results.

> I didn't even know this is possible. How can this be checked?
Maybe `cell.getGeometry().relative`;

Reply:
How do I mock a Javascript function from the python test?

Reply:
There is not a implementation for that I think.

```python
def mocked_js_random(graph):
    graph.selenium.execute_script("_mocked_js_random_original = Math.random;")
    # 4: Chosen by fair dice roll (d6). Guaranteed to be random.
    # Math.random => [0, 1)
    # (4 - 1) / 6 = 0.5
    graph.selenium.execute_script("Math.random = function(){ return 0.5; }")
    try:
        yield
    finally:
        graph.selenium.execute_script("Math.random = _mocked_js_random_original;")
```